### ISSUE TITLE
texture problem

### ISSUE THREAD

**Issue by AzizAydin97** [TencentARC/InstantMesh#67]
thanks for the great work. 
i have a problem when I am running the code on colab environment . the results given doesn't include the texture of the body. however , when I use the code on huggingface it give me the results with a textured 3d structure as shown in the below figure. 
![Screenshot 2024-05-05 213932](https://github.com/TencentARC/InstantMesh/assets/94117798/a3b8ff48-a226-4396-9bc1-cc8c5e11592e)

when I use the code on colab it only give me this result 
![image](https://github.com/TencentARC/InstantMesh/assets/94117798/fb0d06a6-5018-4ba4-9161-a62ee575d6dd)
which has no texture.

please can you help me 

thanks

**Comment by AzizAydin97** [comment]
please i would appreciate your respond

**Comment by bluestyle97** [comment]
How did you visualize the mesh? The huggingface demo exports a mesh with vertex color, while the colab script exports a mesh with texture map. Do you have the `.mtl` file and `.png` texture map under the same path with the `.obj' mesh?

**Comment by AzizAydin97** [comment]
yes. actually i have the .PNG, .mtl, and .obj files together . they're generated. but i want to get the .obj file with colors and texture  as the output in hugging face. or do i have to combine the .obj file alongside with the. mtl file ?
i would appreciate your answer

**Comment by AzizAydin97** [comment]
> How did you visualize the mesh? The huggingface demo exports a mesh with vertex color, while the colab script exports a mesh with texture map. Do you have the `.mtl` file and `.png` texture map under the same path with the `.obj' mesh?




yes. actually i have the .PNG, .mtl, and .obj files together . they're generated. but i want to get the .obj file with colors and texture as the output in hugging face. or do i have to combine the .obj file alongside with the. mtl file ?
i would appreciate your answer

**Comment by trueices** [comment]
Originally, obj file does not implemented texture.
For texture embedding, should convert obj to ply or glb.
To convert .obj files to other formats like .ply or .glb can be done with libraries such as trimesh, pyrender, and pygltflib

```
import trimesh

# Load the .obj file
mesh = trimesh.load('model.obj')

# Export to .ply
mesh.export('model.ply')
```

if you need more help, please contact me.

**Comment by AbhinavJangra29** [comment]
hey @trueices thanks for the response, but is there a way we could integrate the obj,mtl and the png file to generate a colored mesh instead. could you provide a simple working code , thanks
